tsdb/head.go

```diff
@@ -373,32 +380,40 @@ func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, opts *HeadOpti
 		opts.SeriesCallback = &noopSeriesLifecycleCallback{}
 	}
 
-	es, err := NewCircularExemplarStorage(opts.NumExemplars, r)
-	if err != nil {
-		return nil, err
+	var em *ExemplarMetrics
```

Comment:
We should avoid this being `nil`. There are cases where ApplyConfig can panic and we are not updating `h.exemplarMetrics` in `ApplyConfig` if we moved from <0 exemplars to >0. So we could start from this being not `nil` from the beginning.

```suggestion
	em := NewExemplarMetrics(r)
```

Reply:
`There are cases where ApplyConfig can panic` can you elaborate?